src/services/facilityService.ts

```diff
@@ -1,13 +1,48 @@
-import { facilities } from '../mockData/mockData';
 import { Facility } from '../typeDefs/gqlTypes';
+import {Facility as PrismaFacility, Contact as PrismaContact } from '@prisma/client';
+
+import prisma from '../db/client';
+
+type FacilityAndRelations = (PrismaFacility & {
+    contact: PrismaContact;
+})
+
+function convertPrismaToGraphQLFacility(input: FacilityAndRelations) {
+    return {
+        id: String(input.id),
+        nameJa: input.nameJa,
+        nameEn: input.nameEn,
+        contact: input.contact,
+        healthcareProfessionals: []
+    } as Facility;
+}
 
 // TODO: add a validation step for incoming parameters
-export const getFacilityById = (id: string) => {
-    const matchingFacility = facilities.find(
-        (location: Facility) => location.id === id
+export const getFacilityById = async (id: string) => {
+    const found = await prisma.facility.findUnique(
+        {where: {
+            id: parseInt(id)
+        },
+        include: {
+            contact: true
+        }}
     );
 
-    return matchingFacility;
+    return found ? convertPrismaToGraphQLFacility(found) : null;
 };
 
-export const getFacilities = () => facilities;
+export const getFacilities = async () => {
+    const facilities = await prisma.facility.findMany(
+        {include: {
+            contact: true
+        }}
+    );
+
+    const ret = Array<Facility>();
+
+    for (let i = 0; i < facilities.length; i++) {
```

Comment:
personal preference nit: could use `facilities.foreach()` instead of `for()`

Reply:
I thought you asked us to be careful with foreach 🤔 

Is there a good rule of thumb for when to use each? I know foreach isn't always run in order

Reply:
sure! 

### use foreach when:
1) you want to do an operation on each record
2) order doesn't matter

### don't use foreach when:
1) you are changing the shape of the data or changing the order or size. You can use map, reduce, etc. 
2) the operation is async
3) you have 10,000+ records. It does have a minor performance hit on huge data sets